### ISSUE TITLE
Questions about the optimization via gradient descend.

### ISSUE THREAD

**Issue by killandy** [kornia/kornia#825]
## ❓ Questions and Help

1. In some augmentation operations, the parameters are tuples which indicates a selection range. However when we update the parameters by gradients, the range start may be larger than the end, how to avoid this or fix it?
The code is :

   ```
   m = nn.Parameter(torch.tensor([0.7, 0.7]))
   print('m', m.cpu().data.numpy())
   mse = nn.MSELoss()
   
   a = KA.ColorJitter(brightness=m, contrast=0, saturation=0, hue=0, same_on_batch=True, p=1)
   optimizer = torch.optim.SGD(a.parameters(), 0.9, momentum=0.9)
   
   for i in range(5):
       print('step:', i)
       save_image(train_images, 'input'+str(i))
       output = a(train_images)
       save_image(output, 'output'+str(i))
       y = mse(train_images, output)
       optimizer.zero_grad()
       y.backward()
       optimizer.step()
       print('m', m.cpu().data.numpy())
   ```

   and got the error:
```
   Exception has occurred: ValueError

   brightness[0] should be smaller than brightness[1] got Parameter containing: tensor([1.0917, 0.7595], requires_grad=True)
```

2. It seems we cannot update the single parameter in some augmentations like ColorJitter:

The code is :
   ```
   m = nn.Parameter(torch.tensor(0.7))
   
   print('m', m.cpu().data.numpy())
   
   mse = nn.MSELoss() 
   
   a = KA.ColorJitter(brightness=m, contrast=0, saturation=0, hue=0, same_on_batch=True, p=1)
   
   optimizer = torch.optim.SGD(a.parameters(), 0.9, momentum=0.9)
   ```

   and got the error:

```
   Exception has occurred: ValueError

   optimizer got an empty parameter list
```

Is there any way we can only update the single parameter (like nn.Parameter(torch.tensor(0.7))) instead of the range(nn.Parameter(torch.tensor([0.7, 0.7])))?

**Comment by shijianjian** [comment]
1. I think nn.Parameter(torch.tensor(0.7)) shall be applicable in current master branch. Can you try master instead? I will test it after back home. 

2. We did not test this use case. Not sure if it works on master. I will test it and fix it very soon. 


Thanks for reporting. :)

**Comment by shijianjian** [comment]
Current master does support single value input, but the gradient graph is broken. It will be fixed in #826.

**Comment by killandy** [comment]
Thanks.
